### ISSUE TITLE
Unable to ray up

### ISSUE THREAD

**Issue by tonychenxyz** [mlfoundations/dclm#69]
When I tried the sample ray config yaml in https://github.com/mlfoundations/dclm/tree/main/baselines, I got the following error

```
>>> ray up my-cluster.yaml

Traceback (most recent call last):
  File "/shared/share_mala/conda_envs/dclm/bin/ray", line 8, in <module>
    sys.exit(main())
  File "/shared/share_mala/conda_envs/dclm/lib/python3.10/site-packages/ray/scripts/scripts.py", line 2615, in main
    return cli()
  File "/shared/share_mala/conda_envs/dclm/lib/python3.10/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/shared/share_mala/conda_envs/dclm/lib/python3.10/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/shared/share_mala/conda_envs/dclm/lib/python3.10/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/shared/share_mala/conda_envs/dclm/lib/python3.10/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/shared/share_mala/conda_envs/dclm/lib/python3.10/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/shared/share_mala/conda_envs/dclm/lib/python3.10/site-packages/ray/autoscaler/_private/cli_logger.py", line 856, in wrapper
    return f(*args, **kwargs)
  File "/shared/share_mala/conda_envs/dclm/lib/python3.10/site-packages/ray/scripts/scripts.py", line 1321, in up
    create_or_update_cluster(
  File "/shared/share_mala/conda_envs/dclm/lib/python3.10/site-packages/ray/autoscaler/_private/commands.py", line 275, in create_or_update_cluster
    importer = _NODE_PROVIDERS.get(config["provider"]["type"])
KeyError: 'provider'
```

**Comment by GeorgiosSmyrnis** [comment]
Hello @tonychenxyz !

I assume you want to run the cluster on AWS EC2 instances. In this case, you can use the `provider` field to specify the region for this - here is another example script: https://github.com/mlfoundations/open_lm/blob/main/open_lm/datapreprocess/ray/ray_cluster_configs/cluster_west.yaml

Make sure to check the instance type and its associated costs if you do so!

**Comment by tonychenxyz** [comment]
Thanks very much!

I got another error
`jsonschema.exceptions.ValidationError: Additional properties are not allowed ('autoscaling_mode', 'head_node', 'initial_workers', 'worker_nodes' were unexpected)`

**Comment by GeorgiosSmyrnis** [comment]
HI @tonychenxyz ,

I think the above fields can be removed from the yaml without issue.

**Comment by tonychenxyz** [comment]
I removed these fields and got another error. Here's the log

```
(/shared/share_mala/conda_envs/dclm) (live)hc3295@researchgpu05:/shared/share_mala/hc3295$ ray up my-cluster.yaml 
Cluster: my-cluster

2024-09-06 17:40:48,488 INFO util.py:382 -- setting max workers for head node type to 0
2024-09-06 17:40:48,488 INFO util.py:386 -- setting max workers for ray.worker.default to 10
Checking AWS environment settings
AWS config
  IAM Profile: ray-autoscaler-v1 [default]
  EC2 Key pair (all available node types): ray-autoscaler_us-west-2 [default]
  VPC Subnets (all available node types): subnet-0c471f450e216b388, subnet-01911dfdc571936af, subnet-02aa8e450458ee6f6, subnet-0dcfec457d6c05c12 [default]
  EC2 Security groups (all available node types): sg-0cd81799530d1ddac [default]
  EC2 AMI (all available node types): ami-0387d929287ab193e [dlami]

No head node found. Launching a new cluster. Confirm [y/N]: y

Usage stats collection is enabled. To disable this, add `--disable-usage-stats` to the command that starts the cluster, or run the following command: `ray disable-usage-stats` before starting the cluster. See https://docs.ray.io/en/master/cluster/usage-stats.html for more details.

Acquiring an up-to-date head node
  Launched 1 nodes [subnet_id=subnet-0c471f450e216b388]
    Launched instance i-0c683a4a2e90949eb [state=pending, info=pending]
  Launched a new head node
  Fetching the new head node
  
<1/1> Setting up head node
  Prepared bootstrap config
  New status: waiting-for-ssh
  [1/7] Waiting for SSH to become available
    Running `uptime` as a test.
    Fetched IP: 35.85.255.43
ssh: connect to host 35.85.255.43 port 22: Connection timed out
    SSH still not available (SSH command failed.), retrying in 5 seconds.
ssh: connect to host 35.85.255.43 port 22: Connection timed out
    SSH still not available (SSH command failed.), retrying in 5 seconds.
ssh: connect to host 35.85.255.43 port 22: Connection timed out
    SSH still not available (SSH command failed.), retrying in 5 seconds.
ssh: connect to host 35.85.255.43 port 22: Connection timed out
    SSH still not available (SSH command failed.), retrying in 5 seconds.
ssh: connect to host 35.85.255.43 port 22: Connection refused
    SSH still not available (SSH command failed.), retrying in 5 seconds.
Warning: Permanently added '35.85.255.43' (ECDSA) to the list of known hosts.
 21:42:36 up 1 min,  1 user,  load average: 1.93, 0.51, 0.18
Shared connection to 35.85.255.43 closed.
    Success.
  Updating cluster configuration. [hash=5d6e8688e52fadd6e06846d67cfa1cb740d4299e]
  New status: syncing-files
  [2/7] Processing file mounts
Shared connection to 35.85.255.43 closed.
Shared connection to 35.85.255.43 closed.
  [3/7] No worker file mounts to sync
  New status: setting-up
  [4/7] No initialization commands to run.
  [5/7] Initializing command runner
Shared connection to 35.85.255.43 closed.
latest: Pulling from rayproject/ray
762bedf4b1b7: Pull complete 
f2fd58840372: Pull complete 
2c1d0fc2a67a: Pull complete 
4f4fb700ef54: Pull complete 
381607b7196f: Pull complete 
68fa7204c1d1: Pull complete 
f5f9979a537e: Pull complete 
a39e97828480: Pull complete 
c077d10811e0: Pull complete 
Digest: sha256:cca3f2a182427a39404eca89dd1676f1d437b58057df2723b270b56e9dc63b55
Status: Downloaded newer image for rayproject/ray:latest
docker.io/rayproject/ray:latest
Shared connection to 35.85.255.43 closed.
Shared connection to 35.85.255.43 closed.
Shared connection to 35.85.255.43 closed.
Shared connection to 35.85.255.43 closed.
Shared connection to 35.85.255.43 closed.
NVIDIA-SMI has failed because it couldn't communicate with the NVIDIA driver. Make sure that the latest NVIDIA driver is installed and running.

Shared connection to 35.85.255.43 closed.
2024-09-06 17:44:46,177 WARNING command_runner.py:874 -- Nvidia Container Runtime is present, but no GPUs found.
ec73c11e8c5ec991691b1962aa4a19a2146cca4a1567aa6f30005d48ab5c63db
Shared connection to 35.85.255.43 closed.
Shared connection to 35.85.255.43 closed.
sending incremental file list
ray_bootstrap_config.yaml

sent 1,025 bytes  received 35 bytes  2,120.00 bytes/sec
total size is 2,331  speedup is 2.20
Shared connection to 35.85.255.43 closed.
Shared connection to 35.85.255.43 closed.
sending incremental file list
ray_bootstrap_key.pem

sent 1,404 bytes  received 35 bytes  959.33 bytes/sec
total size is 1,674  speedup is 1.16
Shared connection to 35.85.255.43 closed.
Shared connection to 35.85.255.43 closed.
  [6/7] Running setup commands
    (0/3) pip install -U "ray[default]"
Requirement already satisfied: ray[default] in ./anaconda3/lib/python3.9/site-packages (2.35.0)
Requirement already satisfied: click>=7.0 in ./anaconda3/lib/python3.9/site-packages (from ray[default]) (8.1.7)
Requirement already satisfied: filelock in ./anaconda3/lib/python3.9/site-packages (from ray[default]) (3.13.1)
Requirement already satisfied: jsonschema in ./anaconda3/lib/python3.9/site-packages (from ray[default]) (4.17.3)
Requirement already satisfied: msgpack<2.0.0,>=1.0.0 in ./anaconda3/lib/python3.9/site-packages (from ray[default]) (1.0.7)
Requirement already satisfied: packaging in ./anaconda3/lib/python3.9/site-packages (from ray[default]) (23.0)
Requirement already satisfied: protobuf!=3.19.5,>=3.15.3 in ./anaconda3/lib/python3.9/site-packages (from ray[default]) (3.20.3)
Requirement already satisfied: pyyaml in ./anaconda3/lib/python3.9/site-packages (from ray[default]) (6.0.1)
Requirement already satisfied: aiosignal in ./anaconda3/lib/python3.9/site-packages (from ray[default]) (1.3.1)
Requirement already satisfied: frozenlist in ./anaconda3/lib/python3.9/site-packages (from ray[default]) (1.4.1)
Requirement already satisfied: requests in ./anaconda3/lib/python3.9/site-packages (from ray[default]) (2.31.0)
Requirement already satisfied: aiohttp>=3.7 in ./anaconda3/lib/python3.9/site-packages (from ray[default]) (3.9.5)
Requirement already satisfied: aiohttp-cors in ./anaconda3/lib/python3.9/site-packages (from ray[default]) (0.7.0)
Requirement already satisfied: colorful in ./anaconda3/lib/python3.9/site-packages (from ray[default]) (0.5.5)
Requirement already satisfied: py-spy>=0.2.0 in ./anaconda3/lib/python3.9/site-packages (from ray[default]) (0.3.14)
Requirement already satisfied: opencensus in ./anaconda3/lib/python3.9/site-packages (from ray[default]) (0.11.3)
Requirement already satisfied: pydantic!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.*,!=2.4.*,<3 in ./anaconda3/lib/python3.9/site-packages (from ray[default]) (2.5.0)
Requirement already satisfied: prometheus-client>=0.7.1 in ./anaconda3/lib/python3.9/site-packages (from ray[default]) (0.19.0)
Requirement already satisfied: smart-open in ./anaconda3/lib/python3.9/site-packages (from ray[default]) (6.2.0)
Requirement already satisfied: virtualenv!=20.21.1,>=20.0.24 in ./anaconda3/lib/python3.9/site-packages (from ray[default]) (20.25.3)
Requirement already satisfied: grpcio>=1.32.0 in ./anaconda3/lib/python3.9/site-packages (from ray[default]) (1.60.0)
Requirement already satisfied: memray in ./anaconda3/lib/python3.9/site-packages (from ray[default]) (1.10.0)
Requirement already satisfied: attrs>=17.3.0 in ./anaconda3/lib/python3.9/site-packages (from aiohttp>=3.7->ray[default]) (21.4.0)
Requirement already satisfied: multidict<7.0,>=4.5 in ./anaconda3/lib/python3.9/site-packages (from aiohttp>=3.7->ray[default]) (6.0.5)
Requirement already satisfied: yarl<2.0,>=1.0 in ./anaconda3/lib/python3.9/site-packages (from aiohttp>=3.7->ray[default]) (1.9.4)
Requirement already satisfied: async-timeout<5.0,>=4.0 in ./anaconda3/lib/python3.9/site-packages (from aiohttp>=3.7->ray[default]) (4.0.3)
Requirement already satisfied: annotated-types>=0.4.0 in ./anaconda3/lib/python3.9/site-packages (from pydantic!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.*,!=2.4.*,<3->ray[default]) (0.6.0)
Requirement already satisfied: pydantic-core==2.14.1 in ./anaconda3/lib/python3.9/site-packages (from pydantic!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.*,!=2.4.*,<3->ray[default]) (2.14.1)
Requirement already satisfied: typing-extensions>=4.6.1 in ./anaconda3/lib/python3.9/site-packages (from pydantic!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.*,!=2.4.*,<3->ray[default]) (4.8.0)
Requirement already satisfied: distlib<1,>=0.3.7 in ./anaconda3/lib/python3.9/site-packages (from virtualenv!=20.21.1,>=20.0.24->ray[default]) (0.3.7)
Requirement already satisfied: platformdirs<5,>=3.9.1 in ./anaconda3/lib/python3.9/site-packages (from virtualenv!=20.21.1,>=20.0.24->ray[default]) (3.11.0)
Requirement already satisfied: pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0 in ./anaconda3/lib/python3.9/site-packages (from jsonschema->ray[default]) (0.20.0)
Requirement already satisfied: jinja2>=2.9 in ./anaconda3/lib/python3.9/site-packages (from memray->ray[default]) (3.1.2)
Requirement already satisfied: rich>=11.2.0 in ./anaconda3/lib/python3.9/site-packages (from memray->ray[default]) (13.3.2)
Requirement already satisfied: opencensus-context>=0.1.3 in ./anaconda3/lib/python3.9/site-packages (from opencensus->ray[default]) (0.1.3)
Requirement already satisfied: google-api-core<3.0.0,>=1.0.0 in ./anaconda3/lib/python3.9/site-packages (from opencensus->ray[default]) (1.34.0)
Requirement already satisfied: charset-normalizer<4,>=2 in ./anaconda3/lib/python3.9/site-packages (from requests->ray[default]) (3.3.2)
Requirement already satisfied: idna<4,>=2.5 in ./anaconda3/lib/python3.9/site-packages (from requests->ray[default]) (3.7)
Requirement already satisfied: urllib3<3,>=1.21.1 in ./anaconda3/lib/python3.9/site-packages (from requests->ray[default]) (1.26.19)
Requirement already satisfied: certifi>=2017.4.17 in ./anaconda3/lib/python3.9/site-packages (from requests->ray[default]) (2023.11.17)
Requirement already satisfied: googleapis-common-protos<2.0dev,>=1.56.2 in ./anaconda3/lib/python3.9/site-packages (from google-api-core<3.0.0,>=1.0.0->opencensus->ray[default]) (1.61.0)
Requirement already satisfied: google-auth<3.0dev,>=1.25.0 in ./anaconda3/lib/python3.9/site-packages (from google-api-core<3.0.0,>=1.0.0->opencensus->ray[default]) (2.23.4)
Requirement already satisfied: MarkupSafe>=2.0 in ./anaconda3/lib/python3.9/site-packages (from jinja2>=2.9->memray->ray[default]) (2.1.3)
Requirement already satisfied: markdown-it-py<3.0.0,>=2.2.0 in ./anaconda3/lib/python3.9/site-packages (from rich>=11.2.0->memray->ray[default]) (2.2.0)
Requirement already satisfied: pygments<3.0.0,>=2.13.0 in ./anaconda3/lib/python3.9/site-packages (from rich>=11.2.0->memray->ray[default]) (2.18.0)
Requirement already satisfied: cachetools<6.0,>=2.0.0 in ./anaconda3/lib/python3.9/site-packages (from google-auth<3.0dev,>=1.25.0->google-api-core<3.0.0,>=1.0.0->opencensus->ray[default]) (5.3.2)
Requirement already satisfied: pyasn1-modules>=0.2.1 in ./anaconda3/lib/python3.9/site-packages (from google-auth<3.0dev,>=1.25.0->google-api-core<3.0.0,>=1.0.0->opencensus->ray[default]) (0.3.0)
Requirement already satisfied: rsa<5,>=3.1.4 in ./anaconda3/lib/python3.9/site-packages (from google-auth<3.0dev,>=1.25.0->google-api-core<3.0.0,>=1.0.0->opencensus->ray[default]) (4.7.2)
Requirement already satisfied: mdurl~=0.1 in ./anaconda3/lib/python3.9/site-packages (from markdown-it-py<3.0.0,>=2.2.0->rich>=11.2.0->memray->ray[default]) (0.1.2)
Requirement already satisfied: pyasn1<0.6.0,>=0.4.6 in ./anaconda3/lib/python3.9/site-packages (from pyasn1-modules>=0.2.1->google-auth<3.0dev,>=1.25.0->google-api-core<3.0.0,>=1.0.0->opencensus->ray[default]) (0.5.1)
Shared connection to 35.85.255.43 closed.
    (1/3) pip install boto3
Requirement already satisfied: boto3 in ./anaconda3/lib/python3.9/site-packages (1.26.76)
Requirement already satisfied: botocore<1.30.0,>=1.29.76 in ./anaconda3/lib/python3.9/site-packages (from boto3) (1.29.165)
Requirement already satisfied: jmespath<2.0.0,>=0.7.1 in ./anaconda3/lib/python3.9/site-packages (from boto3) (1.0.1)
Requirement already satisfied: s3transfer<0.7.0,>=0.6.0 in ./anaconda3/lib/python3.9/site-packages (from boto3) (0.6.2)
Requirement already satisfied: python-dateutil<3.0.0,>=2.1 in ./anaconda3/lib/python3.9/site-packages (from botocore<1.30.0,>=1.29.76->boto3) (2.8.2)
Requirement already satisfied: urllib3<1.27,>=1.25.4 in ./anaconda3/lib/python3.9/site-packages (from botocore<1.30.0,>=1.29.76->boto3) (1.26.19)
Requirement already satisfied: six>=1.5 in ./anaconda3/lib/python3.9/site-packages (from python-dateutil<3.0.0,>=2.1->botocore<1.30.0,>=1.29.76->boto3) (1.16.0)
Shared connection to 35.85.255.43 closed.
    (2/3) pip install git+https://github...
bash: syntax error near unexpected token `newline'
Shared connection to 35.85.255.43 closed.
  New status: update-failed
  !!!
  Full traceback: Traceback (most recent call last):
  File "/shared/share_mala/conda_envs/dclm/lib/python3.10/site-packages/ray/autoscaler/_private/updater.py", line 166, in run
    self.do_update()
  File "/shared/share_mala/conda_envs/dclm/lib/python3.10/site-packages/ray/autoscaler/_private/updater.py", line 490, in do_update
    self.cmd_runner.run(cmd, run_env="auto")
  File "/shared/share_mala/conda_envs/dclm/lib/python3.10/site-packages/ray/autoscaler/_private/command_runner.py", line 493, in run
    return self.ssh_command_runner.run(
  File "/shared/share_mala/conda_envs/dclm/lib/python3.10/site-packages/ray/autoscaler/_private/command_runner.py", line 383, in run
    return self._run_helper(final_cmd, with_output, exit_on_fail, silent=silent)
  File "/shared/share_mala/conda_envs/dclm/lib/python3.10/site-packages/ray/autoscaler/_private/command_runner.py", line 298, in _run_helper
    raise click.ClickException(fail_msg) from None
click.exceptions.ClickException: SSH command failed.

  Error message: SSH command failed.
  !!!
  
  Failed to setup head node.
```


Here's my current config

```
cluster_name: my-cluster
min_workers: 1
max_workers: 10
upscaling_speed: 1.0
docker:
  image: "rayproject/ray:latest"
  container_name: "ray_container"
  pull_before_run: True
setup_commands:
  - pip install -U "ray[default]"
  - pip install boto3
head_setup_commands:
  - pip install git+https://github.com/my-repo/dcnlp.git@<branch>#egg=dcnlp --user --extra-index-url https://<PAT>:@github.com/
worker_setup_commands:
  - pip install git+https://github.com/my-repo/dcnlp.git@<branch>#egg=dcnlp --user --extra-index-url https://<PAT>:@github.com/
provider:
    type: aws
    region: us-west-2
    cache_stopped_nodes: False
```

**Comment by GeorgiosSmyrnis** [comment]
Hi @tonychenxyz,

Can you try changing the commands under `head_setup_commands` and `worker_setup_commands` to the following?

```pip install git+https://github.com/mlfoundations/dclm.git```

**Comment by tonychenxyz** [comment]
I still got the error
```
Collecting git+https://github.com/mlfoundations/dclm.git
  Cloning https://github.com/mlfoundations/dclm.git to /tmp/pip-req-build-g0v_dq1l
  Running command git clone --filter=blob:none --quiet https://github.com/mlfoundations/dclm.git /tmp/pip-req-build-g0v_dq1l
  Resolved https://github.com/mlfoundations/dclm.git to commit d615fdd1a5dc1c9455a45229d3895917b69854d4
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [6 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/tmp/pip-req-build-g0v_dq1l/setup.py", line 8, in <module>
          from retrie.retrie import Blacklist
      ModuleNotFoundError: No module named 'retrie'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
Shared connection to 35.85.255.43 closed.
  New status: update-failed
  !!!
  Full traceback: Traceback (most recent call last):
  File "/shared/share_mala/conda_envs/dclm/lib/python3.10/site-packages/ray/autoscaler/_private/updater.py", line 166, in run
    self.do_update()
  File "/shared/share_mala/conda_envs/dclm/lib/python3.10/site-packages/ray/autoscaler/_private/updater.py", line 490, in do_update
    self.cmd_runner.run(cmd, run_env="auto")
  File "/shared/share_mala/conda_envs/dclm/lib/python3.10/site-packages/ray/autoscaler/_private/command_runner.py", line 493, in run
    return self.ssh_command_runner.run(
  File "/shared/share_mala/conda_envs/dclm/lib/python3.10/site-packages/ray/autoscaler/_private/command_runner.py", line 383, in run
    return self._run_helper(final_cmd, with_output, exit_on_fail, silent=silent)
  File "/shared/share_mala/conda_envs/dclm/lib/python3.10/site-packages/ray/autoscaler/_private/command_runner.py", line 298, in _run_helper
    raise click.ClickException(fail_msg) from None
click.exceptions.ClickException: SSH command failed.

  Error message: SSH command failed.
  !!!
  
  Failed to setup head node.
```

My current config is
```
cluster_name: my-cluster
min_workers: 1
max_workers: 10
upscaling_speed: 1.0
docker:
  image: "rayproject/ray:latest"
  container_name: "ray_container"
  pull_before_run: True
setup_commands:
  - pip install -U "ray[default]"
  - pip install boto3
head_setup_commands:
  - pip install git+https://github.com/mlfoundations/dclm.git
worker_setup_commands:
  - pip install git+https://github.com/mlfoundations/dclm.git
provider:
    type: aws
    region: us-west-2
    cache_stopped_nodes: False
```

**Comment by GeorgiosSmyrnis** [comment]
Hi @tonychenxyz,

Can you also try to add `pip install retrie` in the setup commands?

Alternatively, can you also try removing `head_setup_commands` and `worker_setup_commands`, and setting:

```
setup_commands:
    - wget https://repo.anaconda.com/miniconda/Miniconda3-py310_23.3.1-0-Linux-x86_64.sh -O miniconda.sh
    - bash ~/miniconda.sh -f -b -p /tmp/miniconda3/
    - echo 'export PATH="/tmp/miniconda3/bin/:$PATH"' >> ~/.bashrc
    - pip install --upgrade pip setuptools wheel
    - pip install -U "ray[default] @ https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-3.0.0.dev0-cp310-cp310-manylinux2014_x86_64.whl"
    - pip install boto3==1.26.90
    - pip install s3fs==2022.11.0
    - pip install psutil
    - pip install pyarrow
    - pip install 'pandas==2.1.4'
    - pip install git+https://github.com/mlfoundations/open_lm.git
    - git clone https://github.com/mlfoundations/dclm.git
```

This is closer to the one that I've been using (also - whichEC2 instance type are you trying to use?)

**Comment by tonychenxyz** [comment]
The second solution solves the problem. Thank you.
Let me know if I should start a separate issue but I felt it's related:

I successfully launched the ray. 

Then I ran
```
ray attach <your_cluster_config>
cd dcnlp
export PYTHONPATH=$(pwd)
screen -S processing
python3 ray_processing/process.py \
  --source_ref_paths exp_data/datasets/raw_sources/CC_WET_april_2019.json \
  --readable_name c4_v4 \
  --output_dir s3://dcnlp-west/cc_wet_2019_april_baselines/c4_v4/ \
  --config_path baselines/baselines_configs/c4.yaml \
  --source_name cc_april_2019
```

I was told fasttext can not be found. I tried to pip install fasttext both in shell and add pip install fasttext in setup_commands. But I got the following error either way:

```
     File "/tmp/pip-build-env-ic8dq85d/overlay/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 316, in run_command
          self.distribution.run_command(command)
        File "/tmp/pip-build-env-ic8dq85d/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 950, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-ic8dq85d/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 973, in run_command
          cmd_obj.run()
        File "/tmp/pip-build-env-ic8dq85d/overlay/lib/python3.10/site-packages/setuptools/_distutils/command/build.py", line 135, in run
          self.run_command(cmd_name)
        File "/tmp/pip-build-env-ic8dq85d/overlay/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 316, in run_command
          self.distribution.run_command(command)
        File "/tmp/pip-build-env-ic8dq85d/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 950, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-ic8dq85d/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 973, in run_command
          cmd_obj.run()
        File "/tmp/pip-build-env-ic8dq85d/overlay/lib/python3.10/site-packages/setuptools/command/build_ext.py", line 98, in run
          _build_ext.run(self)
        File "/tmp/pip-build-env-ic8dq85d/overlay/lib/python3.10/site-packages/setuptools/_distutils/command/build_ext.py", line 359, in run
          self.build_extensions()
        File "<string>", line 151, in build_extensions
        File "<string>", line 114, in cpp_flag
      RuntimeError: Unsupported compiler -- at least C++17 support is needed!
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for fasttext
Failed to build fasttext
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (fasttext)
```
